model/database.py

```diff
@@ -0,0 +1,108 @@
+import torch
+import os
+from torch.utils.data import Dataset
+import numpy as np
+
+# Access environmental variables
+from dotenv import load_dotenv
+load_dotenv(verbose=True)
+
+# Get Pathname
+input_path = os.environ.get('pathname')
+output_path = os.environ.get('dataname')
+
+class DeNoiseDataset(Dataset):
+    """
+        Dataset that contains the real and generated spectrograms.
+    """
+
+    def __init__(self, data_folder):
+        """
+        Args:
+            data_folder (string): Path to the folder that contains everything.
+        """
+        self.generated = {}
+        self.real = {}
+
+        SPECTROGRAM_GENERATED_PATH = data_folder + '/spectrograms_generated/'
+        SPECTROGRAM_REAL_PATH = data_folder + '/spectrograms_real/'
+
+        # Get all the spectrograms in spectrograms_generated
+        for year in os.listdir(SPECTROGRAM_GENERATED_PATH):
+            for song in os.listdir(SPECTROGRAM_GENERATED_PATH + "/" + year):
+                for file in os.listdir(SPECTROGRAM_GENERATED_PATH + "/" + year + '/' + song):
+                    if file.endswith("20.npy"): # Ignore the end of the song (if less than 20s)
+                        file_name = year + "_" + song + "_" + file
+                        self.generated[file_name] = np.load(SPECTROGRAM_GENERATED_PATH + "/" + year + '/' + song + '/' + file)
```

Comment:
This doesn't work because you are loading the entire dataset into RAM, which is significantly more than 50 000MB.

A better way to do this is using a [generator](https://docs.python.org/3/library/stdtypes.html#generator-types) ([example](https://wiki.python.org/moin/Generators)) type, which can be designed to only load the amount of data you make it load.

Reply:
<img width="210" alt="image" src="https://user-images.githubusercontent.com/71615343/140689690-51c9f427-a082-4e87-b83d-2c16cabdf78d.png">